lib/src/sync-process/sync-message-port.ts

```diff
@@ -158,13 +158,16 @@ export class SyncMessagePort extends EventEmitter {
     message = receiveMessageOnPort(this.port);
     if (message) return message.message;
 
-    assert.equal(Atomics.load(this.buffer, 0), BufferState.Closed);
+    assert.equal(
+      Atomics.and(this.buffer, 0, BufferState.Closed) & BufferState.Closed,
+      BufferState.Closed
+    );
```

Comment:
This now asserts that the old value is either 0x10 or 0x11 and sets the value to 0x10, so that if `receiveMessage()` is called one more time after close we can short-circuit at the first if statement.

Reply:
By "the first if statement", do you mean `if (Atomics.compareExchange(...))`? Since that compares both bits to `BufferState.Closed`, won't it only throw the error if the bits are precisely 0b10?

Reply:
> By "the first if statement", do you mean `if (Atomics.compareExchange(...))`?

Yes.

> won't it only throw the error if the bits are precisely 0b10?

That is the desired behavior. Because if the state is 0b11, we want to receive the very last message that has already been sent right before close, and then set the state from 0b11 to 0b10 so that if `receiveMessage()` is called one more time, it throws as it's closed and we have no more messages.

Reply:
I see, the critical change here is that `Atomics.and()` is _updating_ the value. Putting a critical write operation in an `assert()` call is a bit confusing—although Node doesn't optimize out assertions by default, [users may configure it to](https://github.com/unassert-js/unassert), and readers may assume that the contents of an assertion aren't relevant to the correct behavior of the function.

I'd write this

```suggestion
    const oldState = Atomics.and(this.buffer, 0, BufferState.Closed);
    assert.equal(oldState & BufferState.Closed, BufferState.Closed);
```

...and probably also add a comment indicating why `Atomics.and()` is used here.